lib/ocso_scraper/crawler.rb

```diff
@@ -0,0 +1,79 @@
+require 'httparty'
+require 'uri'
+require 'net/http'
+require 'openssl'
+
+module OcsoScraper
+  class Crawler
+    def self.perform
+      warrants = []
+      ('A'..'Z').each do |first_name_first_letter|
+        ('A'..'Z').each do |last_name_first_letter|
+          last_name_query = last_name_first_letter
+          next_last_name_query = nil
+          continue = true
+          while continue
```

Comment:
in a nested for loop going, with each loop going through the alpha bet, while continue is true, populate results variable using the request method. page_warrants sorts results by last name in the PageParser. 

if next_last_name_query is present then last_name_query get its value, and next becomes nil again.